From 9d351d79363f0cde79066853f6b2069983b27365 Mon Sep 17 00:00:00 2001
From: Dongwon Kim <dongwon.kim@intel.com>
Date: Thu, 8 Dec 2022 12:40:16 -0800
Subject: [PATCH 21/34] ui/gtk: making dmabuf NULL when dmabuf is not valid
 anymore

It is needed to set vc->gfx.guest_fb.dmabuf to NULL to prevent any
further access to the structure once it's released.

Signed-off-by: Dongwon Kim <dongwon.kim@intel.com>
---
 ui/gtk.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/ui/gtk.c b/ui/gtk.c
index b13fb126c5..6f62232137 100644
--- a/ui/gtk.c
+++ b/ui/gtk.c
@@ -657,8 +657,12 @@ static bool gd_has_dmabuf(DisplayChangeListener *dcl)
 static void gd_gl_release_dmabuf(DisplayChangeListener *dcl,
                                  QemuDmaBuf *dmabuf)
 {
+    VirtualConsole *vc = container_of(dcl, VirtualConsole, gfx.dcl);
 #ifdef CONFIG_GBM
     egl_dmabuf_release_texture(dmabuf);
+    if (vc->gfx.guest_fb.dmabuf == dmabuf) {
+        vc->gfx.guest_fb.dmabuf = NULL;
+    }
 #endif
 }
 
-- 
2.30.2

